components/claim/components/ldap_to_email.tsx

```diff
@@ -64,36 +69,37 @@ export default class LDAPToEmail extends React.PureComponent {
             return;
         }
 
-        const confirmPassword = this.passwordConfirmInput.current.value;
+        const confirmPassword = this.passwordConfirmInput.current && this.passwordConfirmInput.current.value;
         if (!confirmPassword || password !== confirmPassword) {
             state.confirmError = Utils.localizeMessage('claim.ldap_to_email.pwdNotMatch', 'Passwords do not match.');
             this.setState(state);
             return;
         }
 
-        state.password = password;
-        state.ldapPassword = ldapPassword;
         this.setState(state);
 
         this.submit(this.props.email, password, '', ldapPassword);
     }
 
-    submit = (loginId, password, token, ldapPassword) => {
-        this.props.switchLdapToEmail(ldapPassword || this.state.ldapPassword, this.props.email, password, token).then(({data, error: err}) => {
-            if (data && data.follow_link) {
+    submit = async (loginId: string, password: string, token: string, ldapPassword: string) => {
+        const res = await this.props.switchLdapToEmail(ldapPassword, this.props.email, password, token);
+        if ('data' in res) {
```

Comment:
This is a slight change in behaviour from the previous code since it'll now skip the else-if statement below if `res` looks like `{data: {}, err: {server_error_id: 'something'}}`.

That said, I don't think that should actually cause problems here since we almost always just return just one of `res.data` or `res.error`, but it's good to be aware of when we might be changing things accidentally.

Reply:
So for this line I need to delete if statement?

Reply:
It should be the same if you combine the outer and inner if statements so it's like `if ('data' in res && res.data.follow_link)` again